include/orb/Orbit.h

```diff
@@ -0,0 +1,391 @@
+/*
+MIT License
+
+Copyright (c) 2020 Pathfinder for Autonomous Navigation (PAN)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+/**
+ * \file Orbit.h
+ * \author Nathan Zimmerberg
+ * \date 10 APR 2020
+ * \brief A class to handle most of the logic related to storing, serializing, propagating, and checking validity of an orbit.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cmath>
+#include <limits>
+#include <lin/core.hpp>
+#include <lin/generators.hpp>
+#include <gnc/config.hpp>
+#include "JacobianHelpers/jacobian_autocoded.h"
+
+#include "geograv.hpp"
+#include "GGM05S.hpp"
+
+namespace orb
+{
+
+GNC_TRACKED_CONSTANT(constexpr static int, PANGRAVORDER,40);
+GNC_TRACKED_CONSTANT(constexpr geograv::Coeff<PANGRAVORDER>, PANGRAVITYMODEL, static_cast<geograv::Coeff<PANGRAVORDER>>(GGM05S));
+
+GNC_TRACKED_CONSTANT(const double, MAXORBITRADIUS, 6378.0E3L+1000.0E3);
+
+GNC_TRACKED_CONSTANT(const double, MINORBITRADIUS, 6378.0E3L);
```

Comment:
Do these constants need to be static as well? I think it's a little weird to have something declared like a global in a header file.

Reply:
I think in C++ const at namespace scope is implicitly static, unless you use extern or something like that. Also the LTO optimizations will remove redundant copies if it gets compiled into multiple translation units.

Reply:
Ah yep right. `const` ensures internal linkage in c++.